processor/src/test/java/org/mapstruct/ap/test/bugs/_1830/BeanMapper.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._1830;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.NullValuePropertyMappingStrategy;
+import org.mapstruct.factory.Mappers;
+
+@Mapper()
+public interface BeanMapper {
+
+    BeanMapper INSTANCE = Mappers.getMapper( BeanMapper.class );
+
+    @Mapping(source = "list", target = "list", nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.CLEAR)
+    BeanDTO map(Bean source, @MappingTarget BeanDTO target);
+
+    @Mapping(source = "bean.list", target = "list", nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.CLEAR)
+    BeanDTO map(NestedBean source, @MappingTarget BeanDTO target);
```

Comment:
In our tests we usually prefer having first the target and then the source. Can we try to keep that consistent?